oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticRequestHandler.java

```diff
@@ -787,7 +787,25 @@ private static QueryBuilder fullTextQuery(String text, String fieldName, PlanRes
             // and could contain other parts like renditions, node name, etc
             return multiMatchQuery.field(fieldName);
         } else {
-            return simpleQueryStringQuery(text).field(fieldName).defaultOperator(Operator.AND);
+            // https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html
+            // simpleQueryStringQuery does not support leading wildcards whereas it's supported by default in queryStringQuery
+            // Not using queryStringQuery by default , since this can have performance impact, also some functional cases break.
```

Comment:
It would be good to know the details (as much as you know them or have a link to documentation) for "this can have performance impact" and "some functional cases break". It is fine if this is documented in the jira issue I think (but then either the commit message or the documentation here needs a link to the jira issue), but it's probably even better if this is a comment in the source code.  I think I know why it would break some tests: simpleQueryString is less strict. But performance impact I'm not sure.

Reply:
@thomasmueller  - thanks for pointing out. I added the details in the comment. Removed the performance impact bit, since that is due to allowingLeadingWildCards, but will impact only those queries that actually search on leading wildcards, but this is in line with the lucene implementation now.